### PR TITLE
Added documentation for options that were missing from %sql Configuration

### LINKED ISSUE #691: some options are undocumented
for example, `autocommit` is undocumented. we need to document all the missing ones: https://jupysql.ploomber.io/en/latest/api/configuration.html

### PR DESCRIPTION
**Pull request by nooodle-soup**
## Describe your changes
Added documentation for options missing from the configuration docs (`autocommit`, `column_local_vars`, `short_errors`)
Reorganized code and documentation for uniformity

## Issue number

Closes #691 

## Checklist before requesting a review

- [x] Performed a self-review of my code
- [x] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)

----
:books: Documentation preview :books:: https://jupysql--867.org.readthedocs.build/en/867/

### CODE REVIEW

**Comment by nooodle-soup** [comment]
Added no-changelog label

**Comment by nooodle-soup** [comment]
Had to remove code example for `short_errors`
It breaks the build during docs build
Could not find a way to make the error pass through

File: src/sql/magic.py [hunk 1]
```diff
@@ -95,38 +96,13 @@ class SqlMagic(Magics, Configurable):
 
     Provides the %%sql magic."""
 
-    displaycon = Bool(
-        default_value=True, config=True, help="Show connection string after execution"
-    )
+    autocommit = Bool(default_value=True, config=True, help="Set autocommit mode")
```
Comment (edublancas) [inline src/sql/magic.py:99]: looks like you ordered them alphabetically (?)

File: doc/api/configuration.md [hunk 2]
```diff
@@ -69,20 +69,18 @@ If you have autopandas set to true, the displaylimit option will not apply. You
 %config SqlMagic.feedback = 0
 ```
 
-## `displaycon`
+## `autocommit`
```
Comment (edublancas) [inline doc/api/configuration.md:72]: looks like you ordered them alphabetically (?)

**Review by edublancas — CHANGES_REQUESTED** [review]
please clarify

File: doc/api/configuration.md (hunk 2 above)
Comment (nooodle-soup) [reply doc/api/configuration.md:72]: Yes

File: src/sql/magic.py (hunk 1 above)
Comment (nooodle-soup) [reply src/sql/magic.py:99]: Yes, I ordered both the code and the documentation for uniformity

File: doc/api/configuration.md (hunk 2 above)
Comment (neelasha23) [reply doc/api/configuration.md:72]: should the ordering be handled in another PR? It's a bit difficult to read the new ones added @edublancas

Comment (nooodle-soup) [reply doc/api/configuration.md:72]: I'll do that

**Review by edublancas — APPROVED** [review]

**Comment by edublancas** [comment]
@nooodle-soup good work! please open another PR with the re-arrangement, I think it's a good idea and helps with clarity

**Comment by nooodle-soup** [comment]
Thanks!
Will do!

**Pull request merged**